server/src/main/java/com/arcbees/beestore/server/guice/CrawlerFilter.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2015 ArcBees Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.arcbees.beestore.server.guice;
+
+import java.io.IOException;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.gwtplatform.crawlerservice.server.CrawlServiceServlet;
+
+@Singleton
+public class CrawlerFilter implements Filter {
+    static final String ESCAPED_FRAGMENT = "_escaped_fragment_";
+
+    private final CrawlServiceServlet crawlServlet;
+
+    @Inject
+    CrawlerFilter(CrawlServiceServlet crawlServlet) {
+        this.crawlServlet = crawlServlet;
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain filterChain)
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse resp = (HttpServletResponse) response;
+
+        String queryString = req.getQueryString();
+        if (queryString != null && queryString.contains(ESCAPED_FRAGMENT)) {
```

Comment:
I would check that the host is under *.arcbees.com to limit what can use this crawler service